docs/guides/two_factor_authentication.md

```diff
@@ -0,0 +1,241 @@
+-d '{
+  "email": "user@example.com",
+  "password": "p455w0RD"
+}' \
+-v -w "\n" | jq
+```
+
+The response would be look similar to:
+
+```json
+{
+    "version": "1",
+    "success": true,
+    "data": {
+        "user_id": "usr_01d9m837smphv4kx7xpaxm4dxp",
+        "user": {},
+        "role": null,
+        "pre_authentication_token": "W9Yz6v5j3h9Z2q71NSDOmt8AR5cYw_m8KcIZHUVki_0",
+        "object": "pre_authentication_token",
+        "master_admin": null,
+        "global_role": "super_admin",
+        "account_id": null,
+        "account": null
+    }
+}
+```
+
+Second, calling to the two-factor authentication endpoint `/admin.login_2fa` by using the `pre_authentication_token` which we take from the previous step to form the authorization header, also specify the passcode which was generated by the TOTP authenticator app in the body like below:
+
+2. Login with passcode (or backup_code)
+
+```bash
+curl http://localhost:4000/api/admin/admin.login_2fa \
+-X POST \
+-H "Accept: application/vnd.omisego.v1+json" \
+-H "Authorization: OMGAdmin $(echo -n "user_id:pre_auth_token" | base64 | tr -d '\n')" \
+-H "Content-Type: application/json" \
+-d '{
+  "passcode": "103405"
+}' \
+-v -w "\n" | jq
+```
+
+The response will be look similar to the previous login response, except the `pre_authentication_token`, which will be now a real `authentication_token`:
+
+```json
+{
+    "version": "1",
+    "success": true,
+    "data": {
+        "user_id": "usr_01d9m837smphv4kx7xpaxm4dxp",
+        "user": {},
+        "role": null,
+        "authentication_token": "JAnlHGB3GNDt51TGZeFGoip55gRIdhnefALZHhhdSv8",
+        "object": "authentication_token",
+        "master_admin": null,
+        "global_role": "super_admin",
+        "account_id": null,
+        "account": null
+    }
+}
+```
+
+Now, we can use the `user_id` and `authentication_token` from the response to form the valid authorization header, to be able to access the authenticated APIs.
+
+### Disable 2FA
+
+To disable the two-factor authentication, calling to the api `/me.disable_2fa` as the following:
+
+```bash
+curl http://localhost:4000/api/admin/me.disable_2fa \
+-X POST \
+-H "Accept: application/vnd.omisego.v1+json" \
+-H "Authorization: OMGAdmin $(echo -n "user_id:auth_token" | base64 | tr -d '\n')" \
+-H "Content-Type: application/json" \
+-d '{
+  "passcode": "103405"
+}' \
+-v -w "\n" | jq
+```
+
+If the `passcode` is correct, the response will be look like as following:
+
+```json
+{
+  "version": "1",
+  "success": true,
+  "data": {}
+}
+```
+
+### Note
+
+1. The user's authentication tokens are deleted after enabled or disabled two-factor authentication, so re-login might required.
+
+2. `backup_code` can be used for disable and login.
+
+3. Each `backup_code` is one-time use.
+
+4. `pre_authentication_token` is returned only when the user has enabled two-factor authentication and log in with email and password, otherwise the  `authentication_token` is returned instead.
```

Comment:
```suggestion
4. `pre_authentication_token` is returned only when the user has two-factor authentication enabled, otherwise the `authentication_token` is returned right after the user logs in with username and password.
```